modules/core/src/command_line_parser.cpp

```diff
@@ -134,22 +135,25 @@ void CommandLineParser::getByIndex(int index, bool space_delete, int type, void*
                 String v = impl->data[i].def_value;
                 if (space_delete == true) v = impl->cat_string(v);
 
-                // it is an error if we just got the default value here
-                if(v.empty())
-                    break;
-
+                // the key was neither specified nor has it a default value
+                if(v.empty()) {
```

Comment:
There are many existed application that uses **optional** string parameters with default empty string.
Also empty value can be passed via command line, see test code below.
Anyway empty string is not an error. There are no significant reasons to fordid this value. Also code with additional `.has()` check is more verbose (removal of check with `.empty()` is wrong).

```
TEST(CommandLineParser, testEmptyStringValue)
{
    static const char * const parameter_keys =
        "{ log       | error,critical | log levels verbosity }"
    ;
    const char* argv[] = {"<bin>", "-log="};
    const int argc = 2;
    cv::CommandLineParser parser(argc, argv, parameter_keys);
    // not sure about this: EXPECT_TRUE(parser.has("log"));
    EXPECT_EQ("", parser.get<String>("log"));
    EXPECT_TRUE(parser.check());
}
```